File: mcp/python/veyronis_mcp.py
#!/usr/bin/env python3
"""
VEYRONIS Python MCP (Model Context Protocol) Bridge
Exposes Veyronis DFIR & threat intelligence tools to any AI model (Ollama, Claude Desktop, OpenAI, LangChain, Cursor).
"""

import sys
import subprocess
import os

def run_mcp_bridge():
    veyronis_bin = os.getenv("VEYRONIS_BIN", "veyronis.exe" if sys.platform == "win32" else "veyronis")
    sys.stderr.write(f"[*] Starting VEYRONIS Python MCP Bridge via {veyronis_bin}...\n")
    
    proc = subprocess.Popen(
        [veyronis_bin, "mcp"],
        stdin=subprocess.PIPE,
        stdout=subprocess.PIPE,
        stderr=sys.stderr,
        text=True,
        bufsize=1
    )
    
    try:
        for line in sys.stdin:
            proc.stdin.write(line)
            proc.stdin.flush()
            out_line = proc.stdout.readline()
            if out_line:
                sys.stdout.write(out_line)
                sys.stdout.flush()
    except KeyboardInterrupt:
        proc.terminate()

if __name__ == "__main__":
    run_mcp_bridge()

File: mcp/php/veyronis_mcp.php
<?php
/**
 * VEYRONIS PHP MCP (Model Context Protocol) Bridge
 * Enables PHP backends and AI agents to invoke Veyronis analysis tools via JSON-RPC.
 */

$binName = (PHP_OS_FAMILY === 'Windows') ? 'veyronis.exe' : 'veyronis';
if ($envBin = getenv('VEYRONIS_BIN')) {
    $binName = $envBin;
}

fwrite(STDERR, "[*] Starting VEYRONIS PHP MCP Bridge via {$binName}...\n");

$descriptors = [
    0 => ["pipe", "r"], // stdin
    1 => ["pipe", "w"], // stdout
    2 => STDERR          // stderr
];

$process = proc_open("{$binName} mcp", $descriptors, $pipes);

if (!is_resource($process)) {
    fwrite(STDERR, "[-] Failed to launch Veyronis process.\n");
    exit(1);
}

// Forward standard input to Veyronis and stream responses back
while (($line = fgets(STDIN)) !== false) {
    fwrite($pipes[0], $line);
    fflush($pipes[0]);
    $response = fgets($pipes[1]);
    if ($response !== false) {
        fwrite(STDOUT, $response);
        fflush(STDOUT);
    }
}

fclose($pipes[0]);
fclose($pipes[1]);
proc_close($process);
